src/models/room.ts

```diff
@@ -2141,9 +2148,19 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
             };
         }
 
-        // We've exhausted all scenarios, can safely assume that this event should live in the room timeline only
+        const unsigned = event.getUnsigned();
+        if (typeof unsigned["org.matrix.msc4023.thread_id"] === "string") {
+            return {
+                shouldLiveInRoom: false,
+                shouldLiveInThread: true,
+                threadId: unsigned["org.matrix.msc4023.thread_id"],
+            };
+        }
+
+        // We've exhausted all scenarios,
+        // we cannot assume that it lives in the main timeline as this may be a relation for an unknown thread
```

Comment:
Maybe adding a comment saying that it's better to not add the event rather thna adding it to an incorrect place as it can cause stuck notifs